[BUG] prophet_model.py wrong code duplication

**Describe the bug**
The new seasonality feature in prophet_model.py has an error

![image](https://user-images.githubusercontent.com/15861245/144493339-8adf2c4b-b8f7-4045-888e-67fbaa2a47a7.png)

In addition, I think that the freq.startswith() should be .endswith() at least for the minutes since when I check the 
the time series it gives "30T" as freq_str attribute

**To Reproduce**
Steps to reproduce the behavior, preferably code snippet.

**Expected behavior**
A clear and concise description of what you expected to happen.

**System (please complete the following information):**
 - Python version: [e.g. 3.7]
 - darts version [e.g. 0.5.0]

**Additional context**
Add any other context about the problem here.

Hi @hrzn  thank you for your quick response, I installed it yesterday using `pip install u8darts[all]`, but it still seems that the version is the right one.
![image](https://user-images.githubusercontent.com/15861245/144578579-2a1e2723-9474-405e-8828-8bd749256888.png)
It could the in the way I install the darts

I used the u8darts[all], because I had problems with the `pip install darts` . The prophet could not be built properly and this made the installation to crash. 

System info:
Microsoft Windows 10 Enterprise
Python 3.8.12

I tried to make a new env and install it using `pip install darts` and this is the error when I try it:

Firstly, it fails to build prophet, then it tries to install all the packages and then it fails due to the prophet

Building wheels for collected packages: prophet
  Building wheel for prophet (setup.py) ... error
  ERROR: Command errored out with exit status 1:
   command: 'C:\PROGRA~1\ANACON~1\envs\py38_darts\python.exe' -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\i00202449\\AppData\\Local\\Temp\\pip-install-4sdyy9oz\\prophet_f9b1a33479e843969bb3515d4cc9256a\\setup.py'"'"'; __file__='"'"'C:\\Users\\i00202449\\AppData\\Local\\Temp\\pip-install-4sdyy9oz\\prophet_f9b1a33479e843969bb3515d4cc9256a\\setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d 'C:\Users\i00202449\AppData\Local\Temp\pip-wheel-5g4sjoaw'
       cwd: C:\Users\i00202449\AppData\Local\Temp\pip-install-4sdyy9oz\prophet_f9b1a33479e843969bb3515d4cc9256a\
  Complete output (44 lines):
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build\lib
  creating build\lib\prophet
  creating build\lib\prophet\stan_model
  Traceback (most recent call last):
    File "<string>", line 1, in <module>
    File "C:\Users\i00202449\AppData\Local\Temp\pip-install-4sdyy9oz\prophet_f9b1a33479e843969bb3515d4cc9256a\setup.py", line 123, in <module>
      setup(
    File "C:\PROGRA~1\ANACON~1\envs\py38_darts\lib\site-packages\setuptools\__init__.py", line 153, in setup
      return distutils.core.setup(**attrs)
    File "C:\PROGRA~1\ANACON~1\envs\py38_darts\lib\distutils\core.py", line 148, in setup
      dist.run_commands()
    File "C:\PROGRA~1\ANACON~1\envs\py38_darts\lib\distutils\dist.py", line 966, in run_commands
      self.run_command(cmd)
    File "C:\PROGRA~1\ANACON~1\envs\py38_darts\lib\distutils\dist.py", line 985, in run_command
      cmd_obj.run()
    File "C:\PROGRA~1\ANACON~1\envs\py38_darts\lib\site-packages\wheel\bdist_wheel.py", line 299, in run
      self.run_command('build')
    File "C:\PROGRA~1\ANACON~1\envs\py38_darts\lib\distutils\cmd.py", line 313, in run_command
      self.distribution.run_command(command)
    File "C:\PROGRA~1\ANACON~1\envs\py38_darts\lib\distutils\dist.py", line 985, in run_command
      cmd_obj.run()
    File "C:\PROGRA~1\ANACON~1\envs\py38_darts\lib\distutils\command\build.py", line 135, in run
      self.run_command(cmd_name)
    File "C:\PROGRA~1\ANACON~1\envs\py38_darts\lib\distutils\cmd.py", line 313, in run_command
      self.distribution.run_command(command)
    File "C:\PROGRA~1\ANACON~1\envs\py38_darts\lib\distutils\dist.py", line 985, in run_command
      cmd_obj.run()
    File "C:\Users\i00202449\AppData\Local\Temp\pip-install-4sdyy9oz\prophet_f9b1a33479e843969bb3515d4cc9256a\setup.py", line 48, in run
      build_models(target_dir)
    File "C:\Users\i00202449\AppData\Local\Temp\pip-install-4sdyy9oz\prophet_f9b1a33479e843969bb3515d4cc9256a\setup.py", line 36, in build_models
      from prophet.models import StanBackendEnum
    File "C:\Users\i00202449\AppData\Local\Temp\pip-install-4sdyy9oz\prophet_f9b1a33479e843969bb3515d4cc9256a\prophet\__init__.py", line 8, in <module>
      from prophet.forecaster import Prophet
    File "C:\Users\i00202449\AppData\Local\Temp\pip-install-4sdyy9oz\prophet_f9b1a33479e843969bb3515d4cc9256a\prophet\forecaster.py", line 17, in <module>
      from prophet.make_holidays import get_holiday_names, make_holidays_df
    File "C:\Users\i00202449\AppData\Local\Temp\pip-install-4sdyy9oz\prophet_f9b1a33479e843969bb3515d4cc9256a\prophet\make_holidays.py", line 14, in <module>
      import prophet.hdays as hdays_part2
    File "C:\Users\i00202449\AppData\Local\Temp\pip-install-4sdyy9oz\prophet_f9b1a33479e843969bb3515d4cc9256a\prophet\hdays.py", line 13, in <module>
      from convertdate.islamic import from_gregorian, to_gregorian
  ModuleNotFoundError: No module named 'convertdate'
  ----------------------------------------
  ERROR: Failed building wheel for prophet
  Running setup.py clean for prophet
Failed to build prophet
Installing collected packages: urllib3, pyasn1, idna, charset-normalizer, rsa, requests, pymeeus, pyasn1-modules, oauthlib, cachetools, ujson, threadpoolctl, requests-oauthlib, patsy, korean-lunar-calendar, joblib, importlib-metadata, hijri-converter, google-auth, ephem, Cython, convertdate, werkzeug, typing-extensions, tqdm, tensorboard-plugin-wit, tensorboard-data-server, statsmodels, setuptools-git, scikit-learn, pystan, protobuf, markdown, LunarCalendar, holidays, grpcio, google-auth-oauthlib, cmdstanpy, absl-py, xarray, torch, tensorboard, prophet, pmdarima, lightgbm, filterpy, darts
    Running setup.py install for prophet ... error
    ERROR: Command errored out with exit status 1:
     command: 'C:\PROGRA~1\ANACON~1\envs\py38_darts\python.exe' -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\i00202449\\AppData\\Local\\Temp\\pip-install-4sdyy9oz\\prophet_f9b1a33479e843969bb3515d4cc9256a\\setup.py'"'"'; __file__='"'"'C:\\Users\\i00202449\\AppData\\Local\\Temp\\pip-install-4sdyy9oz\\prophet_f9b1a33479e843969bb3515d4cc9256a\\setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record 'C:\Users\i00202449\AppData\Local\Temp\pip-record-53q_rupr\install-record.txt' --single-version-externally-managed --compile --install-headers 'C:\PROGRA~1\ANACON~1\envs\py38_darts\Include\prophet'
         cwd: C:\Users\i00202449\AppData\Local\Temp\pip-install-4sdyy9oz\prophet_f9b1a33479e843969bb3515d4cc9256a\
    Complete output (122 lines):
    running install
    running build
    running build_py
    creating build
    creating build\lib
    creating build\lib\prophet
    creating build\lib\prophet\stan_model
    Importing plotly failed. Interactive plots will not work.
    INFO:pystan:COMPILING THE C++ CODE FOR MODEL anon_model_f5236004a3fd5b8429270d00efcc0cf9 NOW.
    WARNING:pystan:MSVC compiler is not supported
    stanfit4anon_model_f5236004a3fd5b8429270d00efcc0cf9_6649931874010862568.cpp
    c:\program files\anaconda3\envs\py38_darts\lib\site-packages\numpy\core\include\numpy\npy_1_7_deprecated_api.h(14) : Warning Msg: Using deprecated NumPy API, disable it with #define NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION
    C:\PROGRA~1\ANACON~1\envs\py38_darts\lib\site-packages\pystan\stan\lib\stan_math\stan/math/prim/mat/meta/seq_view.hpp(145): warning C4267: 'return': conversion from 'size_t' to 'int', possible loss of data
    C:\PROGRA~1\ANACON~1\envs\py38_darts\lib\site-packages\pystan\stan\lib\stan_math\stan/math/prim/mat/fun/multiply_lower_tri_self_transpose.hpp(19): warning C4244: 'initializing': conversion from 'Eigen::EigenBase<Derived>::Index' to 'int', possible loss of data
            with
            [
                Derived=Eigen::Matrix<double,-1,-1,0,-1,-1>
            ]
    C:\PROGRA~1\ANACON~1\envs\py38_darts\lib\site-packages\pystan\stan\lib\stan_math\stan/math/prim/mat/fun/multiply_lower_tri_self_transpose.hpp(27): warning C4244: 'initializing': conversion from 'Eigen::EigenBase<Derived>::Index' to 'int', possible loss of data
            with
            [
                Derived=Eigen::Matrix<double,-1,-1,0,-1,-1>
            ]
    C:\PROGRA~1\ANACON~1\envs\py38_darts\lib\site-packages\pystan\stan\lib\stan_math\stan/math/prim/mat/fun/csr_u_to_z.hpp(28): warning C4267: 'argument': conversion from 'size_t' to 'int', possible loss of data
    C:\PROGRA~1\ANACON~1\envs\py38_darts\lib\site-packages\pystan\stan\lib\stan_math\stan/math/prim/mat/fun/matrix_exp_action_handler.hpp(118): warning C4244: 'initializing': conversion from 'double' to 'int', possible loss of data
    C:\PROGRA~1\ANACON~1\envs\py38_darts\lib\site-packages\pystan\stan\lib\stan_math\stan/math/prim/mat/fun/to_matrix.hpp(136): warning C4267: 'initializing': conversion from 'size_t' to 'int', possible loss of data
    C:\PROGRA~1\ANACON~1\envs\py38_darts\lib\site-packages\pystan\stan\lib\stan_math\stan/math/prim/mat/fun/to_row_vector.hpp(31): warning C4267: 'initializing': conversion from 'size_t' to 'int', possible loss of data
    C:\PROGRA~1\ANACON~1\envs\py38_darts\lib\site-packages\pystan\stan\lib\stan_math\stan/math/prim/mat/fun/to_vector.hpp(31): warning C4267: 'initializing': conversion from 'size_t' to 'int', possible loss of data
    C:\PROGRA~1\ANACON~1\envs\py38_darts\lib\site-packages\pystan\stan\lib\stan_math\stan/math/prim/mat/fun/welford_covar_estimator.hpp(31): warning C4244: 'return': conversion from 'double' to 'int', possible loss of data
    C:\PROGRA~1\ANACON~1\envs\py38_darts\lib\site-packages\pystan\stan\lib\stan_math\stan/math/prim/mat/fun/welford_var_estimator.hpp(31): warning C4244: 'return': conversion from 'double' to 'int', possible loss of data
    C:\PROGRA~1\ANACON~1\envs\py38_darts\lib\site-packages\pystan\stan\src\stan/io/dump.hpp(243): warning C4101: 'exc': unreferenced local variable
    C:\PROGRA~1\ANACON~1\envs\py38_darts\lib\site-packages\pystan\stan\src\stan/io/dump.hpp(270): warning C4101: 'exc': unreferenced local variable
    C:\PROGRA~1\ANACON~1\envs\py38_darts\lib\site-packages\pystan\stan\src\stan/io/dump.hpp(284): warning C4101: 'exc': unreferenced local variable
    C:\PROGRA~1\ANACON~1\envs\py38_darts\lib\site-packages\pystan\stan\lib\stan_math\stan/math/rev/mat/meta/operands_and_partials.hpp(42): warning C4267: 'return': conversion from 'size_t' to 'int', possible loss of data
    C:\PROGRA~1\ANACON~1\envs\py38_darts\lib\site-packages\pystan\stan\lib\stan_math\stan/math/rev/mat/meta/operands_and_partials.hpp(155): warning C4267: 'return': conversion from 'size_t' to 'int', possible loss of data
    C:\PROGRA~1\ANACON~1\envs\py38_darts\lib\site-packages\pystan\stan\lib\stan_math\stan/math/rev/scal/fun/bessel_first_kind.hpp(19): warning C4244: 'argument': conversion from 'double' to 'int', possible loss of data
    C:\PROGRA~1\ANACON~1\envs\py38_darts\lib\site-packages\pystan\stan\lib\stan_math\stan/math/rev/scal/fun/bessel_first_kind.hpp(20): warning C4244: 'argument': conversion from 'double' to 'int', possible loss of data
    C:\PROGRA~1\ANACON~1\envs\py38_darts\lib\site-packages\pystan\stan\lib\stan_math\stan/math/rev/scal/fun/bessel_second_kind.hpp(19): warning C4244: 'argument': conversion from 'double' to 'int', possible loss of data
    C:\PROGRA~1\ANACON~1\envs\py38_darts\lib\site-packages\pystan\stan\lib\stan_math\stan/math/rev/scal/fun/bessel_second_kind.hpp(20): warning C4244: 'argument': conversion from 'double' to 'int', possible loss of data
    C:\PROGRA~1\ANACON~1\envs\py38_darts\lib\site-packages\pystan\stan\lib\stan_math\stan/math/rev/scal/fun/modified_bessel_first_kind.hpp(19): warning C4244: 'argument': conversion from 'double' to 'int', possible loss of data
    C:\PROGRA~1\ANACON~1\envs\py38_darts\lib\site-packages\pystan\stan\lib\stan_math\stan/math/rev/scal/fun/modified_bessel_first_kind.hpp(20): warning C4244: 'argument': conversion from 'double' to 'int', possible loss of data
    C:\PROGRA~1\ANACON~1\envs\py38_darts\lib\site-packages\pystan\stan\lib\stan_math\stan/math/rev/scal/fun/modified_bessel_second_kind.hpp(19): warning C4244: 'argument': conversion from 'double' to 'int', possible loss of data
    C:\PROGRA~1\ANACON~1\envs\py38_darts\lib\site-packages\pystan\stan\lib\stan_math\stan/math/rev/scal/fun/modified_bessel_second_kind.hpp(20): warning C4244: 'argument': conversion from 'double' to 'int', possible loss of data
    C:\PROGRA~1\ANACON~1\envs\py38_darts\lib\site-packages\pystan\stan\lib\stan_math\stan/math/rev/scal/fun/rising_factorial.hpp(18): warning C4244: 'argument': conversion from 'double' to 'int', possible loss of data
    C:\PROGRA~1\ANACON~1\envs\py38_darts\lib\site-packages\pystan\stan\lib\stan_math\stan/math/rev/mat/fun/cholesky_decompose.hpp(86): warning C4244: 'initializing': conversion from 'Eigen::EigenBase<Derived>::Index' to 'int', possible loss of data
            with
            [
                Derived=Eigen::Matrix<stan::math::var,-1,-1,0,-1,-1>
            ]
    C:\PROGRA~1\ANACON~1\envs\py38_darts\lib\site-packages\pystan\stan\lib\stan_math\stan/math/rev/mat/fun/cholesky_decompose.hpp(198): warning C4244: 'initializing': conversion from 'Eigen::EigenBase<Derived>::Index' to 'int', possible loss of data
            with
            [
                Derived=Eigen::Matrix<stan::math::var,-1,-1,0,-1,-1>
            ]
    C:\PROGRA~1\ANACON~1\envs\py38_darts\lib\site-packages\pystan\stan\lib\stan_math\stan/math/rev/mat/fun/log_softmax.hpp(104): warning C4244: 'argument': conversion from 'Eigen::EigenBase<Derived>::Index' to 'int', possible loss of data
            with
            [
                Derived=Eigen::Matrix<stan::math::var,-1,1,0,-1,1>
            ]
    C:\PROGRA~1\ANACON~1\envs\py38_darts\lib\site-packages\pystan\stan\lib\stan_math\stan/math/rev/mat/fun/multiply_lower_tri_self_transpose.hpp(21): warning C4244: 'initializing': conversion from 'Eigen::EigenBase<Derived>::Index' to 'int', possible loss of data
            with
            [
                Derived=Eigen::Matrix<stan::math::var,-1,-1,0,-1,-1>
            ]
    C:\PROGRA~1\ANACON~1\envs\py38_darts\lib\site-packages\pystan\stan\lib\stan_math\stan/math/rev/mat/fun/multiply_lower_tri_self_transpose.hpp(22): warning C4244: 'initializing': conversion from 'Eigen::EigenBase<Derived>::Index' to 'int', possible loss of data
            with
            [
                Derived=Eigen::Matrix<stan::math::var,-1,-1,0,-1,-1>
            ]
    C:\PROGRA~1\ANACON~1\envs\py38_darts\lib\site-packages\pystan\stan\lib\stan_math\stan/math/rev/mat/functor/adj_jac_apply.hpp(410): fatal error C1001: An internal error has occurred in the compiler.
    (compiler file 'msc1.cpp', line 1468)
     To work around this problem, try simplifying or changing the program near the locations listed above.
    Please choose the Technical Support command on the Visual C++
     Help menu, or open the Technical Support help file for more information
    C:\PROGRA~1\ANACON~1\envs\py38_darts\lib\site-packages\pystan\stan\lib\stan_math\stan/math/rev/mat/functor/adj_jac_apply.hpp(408): note: while compiling class template member function 'auto stan::math::adj_jac_vari<F,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>::operator ()(const Eigen::Matrix<stan::math::var,-1,1,0,-1,1> &)'
            with
            [
                F=stan::math::internal::ordered_constrain_op
            ]
    C:\PROGRA~1\ANACON~1\envs\py38_darts\lib\site-packages\pystan\stan\lib\stan_math\stan/math/rev/mat/functor/adj_jac_apply.hpp(617): note: see reference to class template instantiation 'stan::math::adj_jac_vari<F,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>' being compiled
            with
            [
                F=stan::math::internal::ordered_constrain_op
            ]
    C:\PROGRA~1\ANACON~1\envs\py38_darts\lib\site-packages\pystan\stan\lib\stan_math\stan/math/rev/mat/fun/ordered_constrain.hpp(88): note: see reference to function template instantiation 'auto stan::math::adj_jac_apply<stan::math::internal::ordered_constrain_op,Eigen::Matrix<stan::math::var,-1,1,0,-1,1>>(const Eigen::Matrix<stan::math::var,-1,1,0,-1,1> &)' being compiled
    C:\PROGRA~1\ANACON~1\envs\py38_darts\lib\site-packages\pystan\stan\lib\stan_math\lib\boost_1.69.0\boost/numeric/odeint/util/bind.hpp(48): note: see reference to class template instantiation 'boost::arg<2>' being compiled
    C:\PROGRA~1\ANACON~1\envs\py38_darts\lib\site-packages\pystan\stan\lib\stan_math\lib\boost_1.69.0\boost/numeric/odeint/util/bind.hpp(47): note: see reference to class template instantiation 'boost::arg<1>' being compiled
    C:\PROGRA~1\ANACON~1\envs\py38_darts\lib\site-packages\pystan\stan\lib\stan_math\lib\boost_1.69.0\boost/math/constants/constants.hpp(264): note: see reference to class template instantiation 'boost::math::constants::detail::constant_half<T>' being compiled
            with
            [
                T=double
            ]
    C:\PROGRA~1\ANACON~1\envs\py38_darts\lib\site-packages\pystan\stan\lib\stan_math\lib\boost_1.69.0\boost/math/constants/constants.hpp(264): note: see reference to function template instantiation 'double boost::math::constants::half<T,boost::math::policies::policy<boost::math::policies::default_policy,boost::math::policies::default_policy,boost::math::policies::default_policy,boost::math::policies::default_policy,boost::math::policies::default_policy,boost::math::policies::default_policy,boost::math::policies::default_policy,boost::math::policies::default_policy,boost::math::policies::default_policy,boost::math::policies::default_policy,boost::math::policies::default_policy,boost::math::policies::default_policy,boost::math::policies::default_policy>>(void) noexcept' being compiled
            with
            [
                T=double
            ]
    C:\PROGRA~1\ANACON~1\envs\py38_darts\lib\site-packages\pystan\stan\lib\stan_math\lib\boost_1.69.0\boost/math/special_functions/detail/bernoulli_details.hpp(82): note: see reference to function template instantiation 'double boost::math::constants::half<double>(void) noexcept' being compiled
    C:\PROGRA~1\ANACON~1\envs\py38_darts\lib\site-packages\pystan\stan\lib\stan_math\lib\boost_1.69.0\boost/math/constants/constants.hpp(277): note: see reference to class template instantiation 'boost::math::constants::detail::constant_pi<T>' being compiled

            with
            [
                T=double
            ]
    C:\PROGRA~1\ANACON~1\envs\py38_darts\lib\site-packages\pystan\stan\lib\stan_math\lib\boost_1.69.0\boost/math/constants/constants.hpp(277): note: see reference to function template instantiation 'double boost::math::constants::pi<T,boost::math::policies::policy<boost::math::policies::default_policy,boost::math::policies::default_policy,boost::math::policies::default_policy,boost::math::policies::default_policy,boost::math::policies::default_policy,boost::math::policies::default_policy,boost::math::policies::default_policy,boost::math::policies::default_policy,boost::math::policies::default_policy,boost::math::policies::default_policy,boost::math::policies::default_policy,boost::math::policies::default_policy,boost::math::policies::default_policy>>(void) noexcept' being compiled
            with
            [
                T=double
            ]
    C:\PROGRA~1\ANACON~1\envs\py38_darts\lib\site-packages\pystan\stan\lib\stan_math\stan/math/prim/scal/fun/constants.hpp(75): note: see reference to function template instantiation 'double boost::math::constants::pi<double>(void) noexcept' being compiled
    C:\PROGRA~1\ANACON~1\envs\py38_darts\lib\site-packages\pystan\stan\lib\stan_math\lib\boost_1.69.0\boost/math/constants/constants.hpp(302): note: see reference to class template instantiation 'boost::math::constants::detail::constant_e<T>' being compiled
            with
            [
                T=double
            ]
    C:\PROGRA~1\ANACON~1\envs\py38_darts\lib\site-packages\pystan\stan\lib\stan_math\lib\boost_1.69.0\boost/math/constants/constants.hpp(302): note: see reference to function template instantiation 'double boost::math::constants::e<T,boost::math::policies::policy<boost::math::policies::default_policy,boost::math::policies::default_policy,boost::math::policies::default_policy,boost::math::policies::default_policy,boost::math::policies::default_policy,boost::math::policies::default_policy,boost::math::policies::default_policy,boost::math::policies::default_policy,boost::math::policies::default_policy,boost::math::policies::default_policy,boost::math::policies::default_policy,boost::math::policies::default_policy,boost::math::policies::default_policy>>(void) noexcept' being compiled
            with
            [
                T=double
            ]
    C:\PROGRA~1\ANACON~1\envs\py38_darts\lib\site-packages\pystan\stan\lib\stan_math\stan/math/prim/scal/fun/constants.hpp(14): note: see reference to function template instantiation 'double boost::math::constants::e<double>(void) noexcept' being compiled
    Internal Compiler Error in C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\BIN\x86_amd64\cl.exe.  You will be prompted to send an error report to Microsoft later.
    error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio 14.0\\VC\\BIN\\x86_amd64\\cl.exe' failed with exit status 2
    ----------------------------------------
ERROR: Command errored out with exit status 1: 'C:\PROGRA~1\ANACON~1\envs\py38_darts\python.exe' -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\i00202449\\AppData\\Local\\Temp\\pip-install-4sdyy9oz\\prophet_f9b1a33479e843969bb3515d4cc9256a\\setup.py'"'"'; __file__='"'"'C:\\Users\\i00202449\\AppData\\Local\\Temp\\pip-install-4sdyy9oz\\prophet_f9b1a33479e843969bb3515d4cc9256a\\setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record 'C:\Users\i00202449\AppData\Local\Temp\pip-record-53q_rupr\install-record.txt' --single-version-externally-managed --compile --install-headers 'C:\PROGRA~1\ANACON~1\envs\py38_darts\Include\prophet' Check the logs for full command output.

--- a/darts/models/forecasting/prophet_model.py
+++ b/darts/models/forecasting/prophet_model.py
@@ -5,6 +5,7 @@ Facebook Prophet
 
 from typing import Optional, Union, List
 
+import re
 import logging
 import prophet
 import numpy as np
@@ -307,34 +308,43 @@ class Prophet(DualCovariatesForecastingModel):
         freq
             frequency string of the underlying TimeSeries's time index (pd.DateTimeIndex.freq_str)
         """
+
+        # this regex extracts all digits from `freq`: exp: '30S' -> 30
+        freq_times = re.findall(r'\d+', freq)
+        freq_times = 1 if not freq_times else int(freq_times[0])
+
+        # this regex extracts all characters and '-' from `freq` and then extracts left string from '-'
+        # exp: 'W-SUN' -> 'W', '30S' -> 'S'
+        freq = ''.join(re.split('[^a-zA-Z-]*', freq)).split('-')[0]
+
         seconds_per_day = 86400
-        if freq == ['A', 'BYS', 'BA', 'RE'] or freq.startswith(('A', 'BYS', 'BA', 'RE-')):  # year
+        if freq in ['A', 'BA', 'Y', 'BY', 'RE'] or freq.startswith(('A', 'BA', 'Y', 'BY', 'RE')):  # year
             days = 365.25
-        elif freq == ['Q', 'BQ', 'REQ'] or freq.startswith(('Q', 'BQ', 'REQ')):  # quarter
+        elif freq in ['Q', 'BQ', 'REQ'] or freq.startswith(('Q', 'BQ', 'REQ')):  # quarter
             days = 3 * 30.4375
         elif freq in ['M', 'BM', 'CBM', 'SM'] or freq.startswith(('M', 'BM', 'BS', 'CBM', 'SM')):  # month
             days = 30.4375
-        elif freq in 'W' or freq.startswith('W-'):  # week
+        elif freq in ['W']:  # week
             days = 7.
-        elif freq in ['B', 'C'] or freq.startswith(('B-', 'C-')):  # business day
+        elif freq in ['B', 'C']:  # business day
             days = 1 * 7/5
-        elif freq in ['D'] or freq.startswith('D-'):  # day
+        elif freq in ['D']:  # day
             days = 1.
-        elif freq in ['H', 'BH', 'CBH'] or freq.startswith(('H', 'BH', 'CBH')):  # hour
+        elif freq in ['H', 'BH', 'CBH']:  # hour
             days = 1/24
-        elif freq in ['T', 'min'] or freq.startswith(('H', 'BH', 'CBH')):  # minute
+        elif freq in ['T', 'min']:  # minute
             days = 1/(24 * 60)
-        elif freq == 'S' or freq.startswith('S'):  # second
+        elif freq in ['S']:  # second
             days = 1/seconds_per_day
-        elif freq in ['L', 'ms'] or freq.startswith(('L', 'ms')):  # millisecond
+        elif freq in ['L', 'ms']:  # millisecond
             days = 1/(seconds_per_day * 10**3)
-        elif freq == ['U', 'us'] or freq.startswith(('U', 'us')):  # microsecond
+        elif freq in ['U', 'us']:  # microsecond
             days = 1/(seconds_per_day * 10**6)
-        elif freq == 'N' or freq.startswith('N'):  # nanosecond
+        elif freq in ['N']:  # nanosecond
             days = 1/(seconds_per_day * 10**9)
         else:
             raise ValueError("freq {} not understood. Please report if you think this is in error.".format(freq))
-        return days
+        return freq_times * days
 
     def _supports_range_index(self) -> bool:
         """Prophet does not support integer range index."""

--- a/darts/tests/test_prophet.py
+++ b/darts/tests/test_prophet.py
@@ -54,15 +54,30 @@ if PROPHET_AVAILABLE:
             perform_full_test = False
 
             test_cases_all = {'A': 12, 'W': 7, 'BM': 12, 'C': 5, 'D': 7, 'MS': 12,
-                              'B': 5, 'H': 24, 'BH': 8, 'Q': 4, 'min': 60, 'S': 60}
+                              'B': 5, 'H': 24, 'BH': 8, 'Q': 4, 'min': 60, 'S': 60, '30S': 60, '24T': 60}
+
             test_cases_fast = {key: test_cases_all[key] for key in ['MS', 'D', 'H']}  # monthly, daily, hourly
 
+            self.helper_test_freq_coversion(test_cases_all)
             test_cases = test_cases_all if perform_full_test else test_cases_fast
             for i, (freq, period) in enumerate(test_cases.items()):
                 if not i:
                     self.helper_test_prophet_model(period=period, freq=freq, compare_all_models=True)
                 else:
                     self.helper_test_prophet_model(period=period, freq=freq, compare_all_models=False)
+
+        def helper_test_freq_coversion(self, test_cases):
+            for freq, period in test_cases.items():
+                ts_sine = tg.sine_timeseries(value_frequency=1 / period, length=3, freq=freq)
+                # this should not raise an error if frequency is known
+                _ = Prophet._freq_to_days(freq=ts_sine.freq_str)
+
+            self.assertAlmostEqual(Prophet._freq_to_days(freq='30S'), 30 * Prophet._freq_to_days(freq='S'),
+                                   delta=10e-9)
+
+            # check bad frequency string
+            with self.assertRaises(ValueError):
+                _ = Prophet._freq_to_days(freq='30SS')
 
         def helper_test_prophet_model(self, period, freq, compare_all_models=False):
             """Test which includes adding custom seasonalities and future covariates. The tests compare the output of

